tardis/model/tests/test_base.py

```diff
@@ -425,20 +387,43 @@ def test_radial_1D_geometry_volume(simulation_verysimple, index, expected):
     ],
 )
 def test_composition_elemental_number_density(
-    simulation_verysimple, index, expected
+    simple_composition, index, expected
 ):
-    sim = simulation_verysimple
-    comp = Composition(
-        sim.model.density,
-        sim.model.abundance,
-        sim.plasma.atomic_mass,
-    )
+    comp = simple_composition
 
     assert_almost_equal(
         comp.elemental_number_density.loc[index], expected, decimal=-2
     )
 
 
+@pytest.mark.parametrize(
+    ("index", "expected"),
+    [
+        ((8, 0), 1.4471412e31),
+        ((16, 10), 2.6820129e30),
+        ((20, 19), 1.3464444e29),
+    ],
+)
+def test_model_state_mass(simple_model_state, index, expected):
+    model_state = simple_model_state
+
+    assert_almost_equal((model_state.mass).loc[index], expected, decimal=-27)
+
+
+@pytest.mark.parametrize(
+    ("index", "expected"),
+    [
+        ((8, 0), 5.4470099e53),
+        ((16, 10), 5.0367073e52),
+        ((20, 19), 2.0231745e51),
+    ],
+)
+def test_model_state_number(simple_model_state, index, expected):
+    model_state = simple_model_state
+
+    assert_almost_equal((model_state.number).loc[index], expected, decimal=-47)
```

Comment:
Why are the decimal parameters negative? I am probably just forgetting the syntax :)

Reply:
The comparisons are being made between numbers in the range 1e50, a position decimal parameter signifies how many digits after the decimal point we need to look at whereas a negative parameter signifies how many digits before the decimal we need to ignore.

Reply:
Okay. I thought that e.g. 1.0124e50 should be compared with decimal=4?